src/lib/Components/PrefetchFlatList.tsx

```diff
@@ -0,0 +1,70 @@
+import { usePrefetch } from "lib/utils/prefetchQuery"
+import React, { Ref, useCallback, useReducer, useRef } from "react"
+import { FlatList, FlatListProps, ViewToken } from "react-native"
+
+export type PrefetchFlatListProps<ItemType> = {
+  prefetchUrlExtractor?: (item?: ItemType) => string | undefined
+  prefetchVariablesExtractor?: (item?: ItemType) => object
```

Comment:
```suggestion
  prefetchQueryUrlExtractor?: (item?: ItemType) => string | undefined
  prefetchQueryVariablesExtractor?: (item?: ItemType) => object
```
**thought:** This is just a personal preference, I don't care much about how long variable names are as long as they are descriptive

Reply:
I think both versions would work. I used the version without `Query` because `PrefetchFlatList` could also work with another implementation of `prefetchUrl` that doesn't use the query to prefetch (e.g. preloading the component). Does that make sense?

Reply:
Yes, of course it makes sense. It was just a personal preference